drools-impact-analysis-graph/drools-impact-analysis-graph-common/src/main/java/org/drools/impact/analysis/graph/ModelToGraphConverter.java

```diff
@@ -58,7 +61,7 @@
                     Map<String, Set<Rule>> reactFieldMap = reactFactMap.computeIfAbsent(patternClass, k -> new HashMap<>());
                     Collection<String> reactOnFields = pattern.getReactOnFields();
                     if (reactOnFields.size() == 0) {
-                        // TODO: confirm mask logic (No constraint vs Failed to analyze property reactivity)
+                        // TODO: Currently reactOnFields.size() == 0 assumes that the pattern doesn't have a constraint (e.g. Person()) so no react
                         Set<Rule> ruleSet = reactFieldMap.computeIfAbsent(NO_REACT, k -> new HashSet<>());
                         ruleSet.add(rule);
                     } else {
```

Comment:
Currently, if reactOnFields.size() == 0, it assumes that the pattern doesn't have a constraint (e.g. Person()) so no react. However, in case of "react all" (e.g. Person(blackBoxMethod())), reactOnFields.size() is also 0. Can we know the difference from Model classes?

Reply:
@tkobayas I need to check how this is implemented in the executable model, but yes I will add this info in the model classes.